ecs-cli/modules/cli/local/network/network.go

```diff
@@ -0,0 +1,217 @@
+// permissions and limitations under the License.
+
+// Package network provides functionality to setup and teardown the ECS local network.
+package network
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/client"
+
+	"github.com/docker/docker/api/types/container"
+
+	"github.com/docker/docker/api/types/filters"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/api/types/network"
+
+	"github.com/docker/docker/api/types"
+)
+
+// IdempotentLocalNetworkStarter groups Docker functions to create the ECS local network and
+// the Local Container Endpoints container if they don't exist.
+type IdempotentLocalNetworkStarter interface {
+	idempotentNetworkCreator
+	idempotentContainerStarter
+}
+
+type idempotentNetworkCreator interface {
+	NetworkInspect(ctx context.Context, networkID string, options types.NetworkInspectOptions) (types.NetworkResource, error)
+	NetworkCreate(ctx context.Context, name string, options types.NetworkCreate) (types.NetworkCreateResponse, error)
+}
+
+type idempotentContainerStarter interface {
+	ContainerList(ctx context.Context, options types.ContainerListOptions) ([]types.Container, error)
+	ContainerCreate(ctx context.Context, config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig, containerName string) (container.ContainerCreateCreatedBody, error)
+	ContainerStart(ctx context.Context, containerID string, options types.ContainerStartOptions) error
+}
+
+// Configuration for the ECS local network.
+const (
+	// EcsLocalNetworkName is the name of the network created for local ECS tasks to join.
+	EcsLocalNetworkName = "ecs-local-network"
+
+	// Range of IP addresses that containers in the network get assigned.
+	ecsLocalNetworkSubnet = "169.254.170.0/24"
+)
+
+// Configuration for the Local Endpoints container.
+const (
+	// Name of the image used to pull from DockerHub, see https://hub.docker.com/r/amazon/amazon-ecs-local-container-endpoints
+	localEndpointsImageName = "amazon/amazon-ecs-local-container-endpoints"
+
+	// Reserved IP address that the container listens to answer requests on metadata, creds, and stats.
+	localEndpointsContainerIpAddr = "169.254.170.2"
+
+	// Name of the container, we need to give it a name so that we don't re-create a container every time we setup.
+	localEndpointsContainerName = "amazon-ecs-local-container-endpoints"
+)
+
+// Configuration for Docker requests
+const (
+	// Wait duration for a response from the Docker daemon before returning an error to the user.
+	dockerTimeout = 30 * time.Second
+)
+
+// Setup creates a user-defined bridge network with a running Local Container Endpoints container. If the network
+// already exists or the container is already running then the operation is a no-op.
+//
+// If there is any unexpected errors, we exit the program with a fatal log.
+func Setup(docker IdempotentLocalNetworkStarter) {
```

Comment:
nit: can we rename the "docker" param(s) to be more specific, like "dockerConfiguration" or "dockerOps"? To differentiate from calling the docker daemon or API directly.